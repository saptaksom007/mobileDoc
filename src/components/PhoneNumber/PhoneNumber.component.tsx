import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import i18n from 'ex-react-native-i18n'
import { Actions as NavAction } from 'navigation/SagaNavigation'

import { Linking } from 'react-native'
import { Action } from 'redux'
import { Medium } from 'components/CustomText/CustomText.component'

const cleanPhoneNumber = (value?: string) =>
  (value && value.split(' ').join('')) || ''

interface Props {
  value?: string
  dispatch?(action: Action): void
}

export class PhoneNumber extends PureComponent<Props> {
  phoneCall = () => {
    const { value, dispatch } = this.props
    const url = `tel:${cleanPhoneNumber(value)}`
    Linking.canOpenURL(url)
      .then((supported: any) => {
        if (!supported && dispatch) {
          dispatch(NavAction.showLocalError(i18n.t('error.default')))
        } else {
          return Linking.openURL(url)
        }
        return Promise.resolve()
      })
      .catch(() => {
        if (dispatch) {
          dispatch(NavAction.showLocalError(i18n.t('error.default')))
        }
      })
  }
  render() {
    const { value } = this.props
    return (
      <>
        <Medium
          style={{
            fontSize: 15,
            lineHeight: 30,
          }}
        >
          {'   '}
        </Medium>
        <Medium
          style={{
            textDecorationLine: 'underline',
            fontSize: 16,
            lineHeight: 35,
          }}
          onLongPress={this.phoneCall}
          onPress={this.phoneCall}
        >
          {cleanPhoneNumber(value)}
        </Medium>
      </>
    )
  }
}

function select(): any {
  return {}
}

export default connect(select)(PhoneNumber)
