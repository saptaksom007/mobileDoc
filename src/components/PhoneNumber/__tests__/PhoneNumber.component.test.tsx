import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { PhoneNumber } from 'components/PhoneNumber/PhoneNumber.component'

it('<PhoneNumber/> render correctly', () => {
  const tree = renderer
    .create(<PhoneNumber value={'0300000000'} dispatch={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
