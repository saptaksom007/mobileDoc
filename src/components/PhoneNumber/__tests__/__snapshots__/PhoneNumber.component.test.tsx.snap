// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneNumber/> render correctly 1`] = `
Array [
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontSize": 16.875,
          "lineHeight": 20,
        },
        Object {
          "fontSize": 15,
          "lineHeight": 30,
        },
      ]
    }
  >
       
  </Text>,
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    onLongPress={[Function]}
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontSize": 16.875,
          "lineHeight": 20,
        },
        Object {
          "fontSize": 16,
          "lineHeight": 35,
          "textDecorationLine": "underline",
        },
      ]
    }
  >
    0300000000
  </Text>,
]
`;
